core/trino-spi/src/main/java/io/trino/spi/eventlistener/TaskDetails.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi.eventlistener;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.math.BigDecimal;
+import java.util.Objects;
+
+public class TaskDetails
+{
+    private final String id;
+    private final String nodeHost;
+    private final String state;
+    private final int queuedSplits;
+    private final int runningSplits;
+    private final int blockedSplits;
+    private final int completedSplits;
+    private final int totalSplits;
+    private final long rowsRead;
+    private final long bytesRead;
+    private final long bytesBuffered;
```

Comment:
Should be peakBufferedDataSize, if we track that, since it is not useful to know bufferedDataSize at the end of a query